rest/route/agent.go

```diff
@@ -671,10 +694,13 @@ func (h *attachFilesHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	if err = entry.Upsert(); err != nil {
-		message := fmt.Sprintf("updating artifact file info for task %s: %v", t.Id, err)
-		grip.Error(message)
-		return gimlet.MakeJSONInternalErrorResponder(errors.New(message))
+		catcher.Errorf("updating artifact file info for task %s: %v", t.Id, err)
 	}
+
+	if catcher.HasErrors() {
```

Comment:
Putting this here means that the new entry update logic can fail while the entry upsert on L696 can still succeed. Is that okay or should we move this check above the upsert?

Reply:
Personally I feel like this new logic is more of a nice-to-have and we shouldn't block the most important operation, which is the entry upsert. I can add a comment to that effect.